### PR TITLE
Refactor ISA amount selection

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: client/src/api/block_builder/input_selection/core/requirement/amount.rs [hunk 1]
```diff
@@ -73,41 +73,109 @@ pub(crate) fn amount_sums(
...
+                    self.outputs_sum += diff;
+                    *self.outputs_sdr.entry(*sdruc.return_address()).or_default() += sdruc.amount();
+                }
+
+                *self.inputs_sdr.entry(*sdruc.return_address()).or_default() += sdruc.amount();
+            }
+
+            self.inputs_sum += input.output.amount();
+            self.newly_selected_inputs.push((input.clone(), None));
+            self.newly_selected_ids.insert(*input.output_id());
```
Comment (Thoralf-M) [inline client/src/api/block_builder/input_selection/core/requirement/amount.rs]: Is this better than HashMap<OutputId, InputSigningData>?

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/requirement/amount.rs]: You mean combining the hashset and the vec ?

Comment (Thoralf-M) [reply client/src/api/block_builder/input_selection/core/requirement/amount.rs]: Yes

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/requirement/amount.rs]: Would have to be `HashMap<OutputId, (InputSigningData, Option<AliasTransition>)>` then, but I'll try yeah

Comment (Thoralf-M) [reply client/src/api/block_builder/input_selection/core/requirement/amount.rs]: Yeah sure, just looks like it would be less error prone to combine them

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/requirement/amount.rs]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**